edx_rest_api_client/client.py

```diff
@@ -5,6 +8,35 @@
 
 
 class EdxRestApiClient(slumber.API):
+
+    @classmethod
+    def get_oauth_access_token(cls, url, client_id, client_secret):
+        """ Retrieves OAuth 2.0 access token using the client credentials grant.
+        :param url: Oauth2 access token endpoint
+        :param client_id: client ID
+        :param client_secret: client secret
+        :return: (str, datetime) -- (access token, expiration datetime)
+        """
+        now = datetime.datetime.now(UTC)
+
+        response = requests.post(
+            url,
+            data={
+                "grant_type": "client_credentials",
+                "client_id": client_id,
+                "client_secret": client_secret,
+            }
+        )
+
+        response = response.json()
+
+        access_token = response.get('access_token')
+        expires_in = response.get('expires_in')
+
+        expires_in_datetime = now + datetime.timedelta(seconds=expires_in) if expires_in else None
+
+        return access_token, expires_in_datetime
+
     def __init__(self, url, signing_key=None, username=None, full_name=None, email=None,
```

Comment:
No need to change this now, but long-term we should move to a model similar to that of `requests` where an auth object is passed in. This constructor's signature has grown a bit unwieldly.